examples/mic_vad_streaming/mic_vad_streaming.py

```diff
@@ -171,6 +217,10 @@ def main(ARGS):
                         help="Path to the language model binary file. Default: lm.binary")
     parser.add_argument('-t', '--trie', default='trie',
                         help="Path to the language model trie file created with native_client/generate_trie. Default: trie")
+    parser.add_argument('-d', '--device', type=int, default=None,
```

Comment:
What happens with `None` as a default value ?

Reply:
The kwargs for PyAudio up top originally had no `device` kwarg -- so in this case PyAudio will fall back to using the default device, which in my case is usually wrong.
I'll add a little more comments around this.
cheers!

Reply:
@jorxster I'm a little bit concerned, default value should work out of the box (i.e. pick up one of the mic of your system, even if it's the wrong one), or you should make the argument required.

Reply:
Thank you @lissyx  -- just to be clear I am not changing the default behaviour of PyAudio.open() as was formerly on line 31. 
You wrote "default value should work out of the box" -- and it will! 
I am only supplying an optional extra argument (as -d or --device) which allows one to specify the device, instead of PyAudio's default. 
I cannot foresee issues here nor how I can improve on this.

Reply:
FYI on my system (Kubuntu 18.04) the device index for PyAudio for my Yeti Blue Microphone seems to be random. It may be a handy handy idea to provide a way of specifying device by name, (simple iterate through devices and find matching name) but that may be beyond the scope of this example module.